raspiraw.c

```diff
@@ -923,12 +896,15 @@ int main(int argc, const char** argv) {
 
 	if (cfg.fps > 0)
 	{
-		// TODO: handle modes different to ov5647 as well
-		int h = get_reg(sensor_mode, 0x380E);
-		int l = get_reg(sensor_mode, 0x380F);
-                int n = ((h<<8) + l) * sensor_mode->framerate / cfg.fps;
-		modReg(sensor_mode, 0x380E, 0, 7, n>>8, EQUAL);
-		modReg(sensor_mode, 0x380F, 0, 7, n&0xFF, EQUAL);
+                int n = 1000000000 / (sensor_mode->line_time_ns * cfg.fps);
+		if (n < sensor_mode->min_vts)
+		{
+			vcos_log_error("WARNING: computed VTS %d smaller than min_vts %d !", 
+					n, sensor_mode->min_vts);    
```

Comment:
In this error case you're probably better off adopting min_vts so that it has a hope of actually running.